connexion/validators.py

```diff
@@ -56,7 +67,7 @@ def validate(self, body: dict):
             )
             raise BadRequestProblem(detail=f"{exception.message}{error_path_msg}")
 
-    async def receive(self) -> t.Optional[t.MutableMapping[str, t.Any]]:
+    async def wrapped_receive(self) -> Receive:
```

Comment:
I changed this so validation is not delayed until the stream is consumed by the application. I think it makes sense for connexion to always validate the body (for certain content types), independent on if it's read in the application or not. This also makes it easier to follow / debug validation, and prevents further middlewares / application logic from being executed for an invalid request.